elks/arch/i86/drivers/char/console-direct.c

```diff
@@ -41,34 +43,45 @@
 
 #define MAXPARMS        28
 
+#ifdef CONFIG_CONSOLE_DUAL
+#define MAX_OUTPUTS     2
+#else
+#define MAX_OUTPUTS     1
+#endif
+
 struct console;
 typedef struct console Console;
 
 struct console {
     int cx, cy;                 /* cursor position */
     void (*fsm)(Console *, int);
+    unsigned char op;
+    unsigned char type;
     unsigned char attr;         /* current attribute */
     unsigned char XN;           /* delayed newline on column 80 */
-    unsigned int vseg;          /* video segment for page */
-    int basepage;               /* start of video ram */
+    unsigned int vseg;          /* vram for this console page */
+    int vseg_offset;            /* vram offset of vseg for this console page */
+    unsigned short w;
+    unsigned short max_col;
+    unsigned short h;
+    unsigned short max_row;
+    unsigned short crtc_base;   /* 6845 CRTC base I/O address */
 #ifdef CONFIG_EMUL_ANSI
     int savex, savey;           /* saved cursor position */
     unsigned char *parmptr;     /* ptr to params */
     unsigned char params[MAXPARMS];     /* ANSI params */
 #endif
 };
 
+static Console *glock;
 static struct wait_queue glock_wait;
-static Console Con[MAX_CONSOLES], *Visible;
-static Console *glock;          /* Which console owns the graphics hardware */
-static unsigned short CCBase;   /* 6845 CRTC base I/O address */
-static int Width, MaxCol, Height, MaxRow;
-static int NumConsoles = MAX_CONSOLES;
-static unsigned char isMDA, isCGA;
+static Console *Visible[MAX_OUTPUTS];
+static Console Con[MAX_CONSOLES];
+static int NumConsoles = 0;
```

Comment:
Remove zero initialization. That will move the data to .bss instead of .data, get automatically zeroed, and reduce the kernel file size by a byte.